core/scale/type_encoder.hpp

```diff
@@ -17,28 +18,28 @@ namespace kagome::common::scale {
   struct TypeEncoder {
     // if required to calculate size, we can add
     // canCalculateSize() and calculateSize(const T& item) methods
-    bool encode(const T &item, Buffer &out) {
+    outcome::result<void> encode(const T &item, Buffer &out) {
       static_assert(std::is_integral<T>(),
                     "Only integral types are supported. You need to define "
                     "your own TypeDecoder specialization for custom type.");
       impl::encodeInteger<T>(item, out);
-      return true;
+      return outcome::success();
     };
   };
 
   template <>
   struct TypeEncoder<bool> {
-    bool encode(bool item, Buffer &out) const {
+    outcome::result<void> encode(bool &&item, Buffer &out) const {
```

Comment:
something weird with types. 

bool&& is not a universal ref, here it says that item is an rvalue.
What is the point?

it is better to pass by value primitive types like ints, bools

Reply:
don't know what made me write like this, maybe it was late)
fixed